app/controller/button/DigitizeButtonController.js

```diff
@@ -0,0 +1,479 @@
+        var feat = evt.feature ? evt.feature : evt.features.getArray();
+        this.prepareRequestParams(feat);
+    },
+
+    /**
+     * Prepares the API request with the given features and calls the `doAjaxRequest` method.
+     * If you need custom handling, just overwrite this method.
+     * @param {array[ol.Feature]|ol.Feature} feat The feature or array of features
+     *   that should be used in the request
+     */
+    prepareRequestParams: function(feat) {
+        var me = this;
+        var view = me.getView();
+        var type = view.getType();
+        var format = new ol.format.GeoJSON({
+            dataProjection: me.map.getView().getProjection().getCode()
+        });
+        var searchParams;
+        var jsonParams;
+        if (type === 'Point') {
+            var feats = me.resultLayer.getSource().getFeatures();
+            if (Ext.isEmpty(feats)) {
+                jsonParams = format.writeFeatures([]);
+            } else {
+                jsonParams = format.writeFeatures(feats.filter(function (f) {
+                    return f.getGeometry().getType() === 'Point' &&
+                      f.get('group') === me.activeGroupIdx;
+                }));
+            }
+            var extent = feat.getGeometry().getExtent();
+            // Netsolver endpoint expect bbox to be sent within a request.
+            // The lower left and upper right coordinates cannot be the same so
+            // we have to apply a small buffer on point geometry to get faked
+            // bbox around the clicked point.
+            var buffered = ol.extent.buffer(extent, 100);
+            searchParams = 'bbox=' + encodeURIComponent(buffered.join(','));
+        }
+        else if (view.getType() === 'LineString') {
+            BasiGX.warn('Please implement your custom handler here for LineStrings');
+        }
+        else if (view.getType() === 'Polygon') {
+            var geoJson = Ext.isArray(feat) ?
+                format.writeFeature(feat[0]) :
+                format.writeFeature(feat);
+            jsonParams = {
+                geometry3857: Ext.JSON.decode(geoJson).geometry
+            };
+        }
+        me.doAjaxRequest(jsonParams, searchParams);
+    },
+
+    /**
+     * Issues an Ext.Ajax.request against the configured endpoint with
+     *   the given params. Returned features will be added to the result
+     *   layer, and the drawn features will be removed. Additionally,
+     *   an event is fired with the response features to allow other
+     *   components to make use of them.
+     * @param {object} jsonParams Object containing the params to send
+     * @param {string} searchParams The serarchParams which will be
+     *   appended to the request url
+     */
+    doAjaxRequest: function (jsonParams, searchParams) {
+        var me = this;
+        var mapComponent = BasiGX.util.Map.getMapComponent();
```

Comment:
I would prefer to pass the `mapComponent` into the constructor and set it as member var. If not passed it could be detected with `guess`.